src/UnitOfWork/OperationConsolidator.php

```diff
@@ -17,8 +18,16 @@ class OperationConsolidator
      *
      * @return Operation[]
      */
-    public function consolidate(array $operations): array
-    {
+    public function consolidate(
+        array $operations,
+        bool $shouldLogUnitOfWorkOperationConsolidation,
+        bool $shouldUseNewConsolidationWithDryRun,
+        bool $shouldUseNewConsolidation
```

Comment:
[M] This is quite unclear at the first glance. What about this?
```suggestion
        OperationsConsolidationMode $operationsConsolidationMode
```
and OperationsConsolidationMode with values:
- NEIGHBORS_ONLY
- UNLIMITED_DRY_RUN
- UNLIMITED

Reply:
In the end, there will be just one parameter - logging so it's just temporary complexity

Reply:
In a case of something outside platform codebase it is usually temporarily permanent, so I have to disagree with making any "temporary" solutions. Let's make this transparently as new feature, where you can choose, how should the consolidator behave.